engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -39,6 +52,7 @@ def verify_product_activation
             verify_extension_activation(product)
           end
         rescue InstanceVerification::Exception => e
+          raise ActionController::TranslatedError.new(e.message) if @system.proxy_byos
```

Comment:
Is a BYOS system actually expected to pass the instance verification or not? 
If not, we could bypass the whole instance verification in that case and also don't need to send `--instance-data` from SUSEConnect. (@rjschwei ?)

Reply:
@digitaltom A BYOS instance is not expected to pass the instance verification. However we have the case where a user may pass a regcode while running on a PAYG instance. As such before doing regcode validation we have to first ensure that PAYG instance verification fails. Otherwise we may put a user into a situation where they end up having an instance counted against their subscription while also paying through the CSP.

Reply:
Could it be that the customer intentionally uses the registration code on PAYG, for example to receive priority support? 
I suggest to clearly document in the code that we do instance verification first and try BYOS with the regcode afterwards because of the use case you described.

Reply:
+1 on doc via comments.

Customers that use PAYG get L1 & L2 support through the CSP. If they want direct support by SUSE for those level they need to switch to BYOS.